src/libcollections/vec_deque.rs

```diff
@@ -1652,13 +1846,16 @@ impl<'a, T: 'a> Iterator for Drain<'a, T> {
 
     #[inline]
     fn next(&mut self) -> Option<T> {
-        self.inner.pop_front()
+        self.iter.next().map(|elt|
+            unsafe {
+                ptr::read(elt as *const _)
```

Comment:
(back then it might've been necessary fwiw -- full &mut to *const coercion is relatively new)